tests/rptest/tests/transactions_test.py

```diff
@@ -678,6 +679,223 @@ def check_pids_overflow_test(self):
         assert num_consumed == should_be_consumed
 
 
+class TransactionsOneTMPartitionDrainingTest(RedpandaTest):
+    topics = (TopicSpec(partition_count=1, replication_factor=3), )
+
+    def __init__(self, test_context):
+        extra_rp_conf = {
+            "enable_leader_balancer": False,
+            "partition_autobalancing_mode": "off",
+            "transaction_coordinator_partitions": 1
+        }
+
+        super(TransactionsOneTMPartitionDrainingTest,
+              self).__init__(test_context=test_context,
+                             extra_rp_conf=extra_rp_conf,
+                             log_level="trace")
+
+        self.admin = Admin(self.redpanda)
+
+    def on_delivery(self, err, msg):
+        assert err == None, msg
+
+    def on_delivery_error(self, err, msg):
+        assert err != None, err
+
+    @cluster(num_nodes=3)
+    def list_draining_transaction_test(self):
+        tx_id = "transaction_0"
+
+        producer = ck.Producer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'transactional.id': tx_id,
+            'transaction.timeout.ms': 10000,
+        })
+
+        producer.init_transactions()
+
+        for i in range(10):
+            producer.begin_transaction()
+            for j in range(10):
+                producer.produce(self.topics[0].name,
+                                 str(i) + "_" + str(j),
+                                 str(j) + "_" + str(j),
+                                 on_delivery=self.on_delivery)
+            producer.commit_transaction()
+
+        draining_ranges = [(0, 100), (500, 1000), (999, 4294967295)]
+        draining_txes = ["tx_1", "tx_2", "tx_3"]
+        admin = Admin(self.redpanda)
+        admin.start_tx_draining(tm_partition=0,
+                                repartitioning_id=100,
+                                tx_ids=draining_txes,
+                                tx_ranges=draining_ranges)
+
+        draining = admin.get_draining_transactions(tm_partition=0)
+
+        assert draining["repartitioning_id"] == 100
+        assert len(draining["hash_ranges"]) == len(draining_ranges)
+        assert set([(x["from"], x["to"])
+                    for x in draining["hash_ranges"]]) == set(draining_ranges)
+        assert len(draining["transactional_ids"]) == len(draining_txes)
+        assert set(draining["transactional_ids"]) == set(draining_txes)
+
+    @cluster(num_nodes=3)
+    @parametrize(drain_type="by_range")
+    @parametrize(drain_type="by_tx_id")
+    def drain_txes_cannot_start_new_test(self, drain_type):
```

Comment:
we should have test for first add_partition (it should be rejected) and for follow up add_partition (it should pass)